File: sort/quick_sort/haskell/QuicksortTest.hs
module QuicksortTest.Main (main) where
import Data.List (sort)

import Quicksort (quicksort)

main :: IO ()
main = do
    if test
    then putStrLn "List sorted correctly"
    else putStrLn "Error: List did not sort correctly"

test :: Bool
test = sort list == quicksort list
  where
    list = [12,3,55,1,2,7,4]

File: sort/quick_sort/java/QuickSortTest.java
import org.junit.Test;
import java.util.Arrays;

public class QuickSortTest {

    @Test
    public void test() {
        int list[] = { -5, 8, 0, 12, 55, 1, -66, 35, 3 };
        int listLength = list.length;
        QuickSort quickSort = new QuickSort();
        quickSort.sort(list, 0, listLength-1);
        for (int i = 0; i < list.length - 1; i++) {
            assert (list[i] <= list[i + 1]);
        }
    }

    @Test
    public void nullList(){
        int list[] = {};
        int expectedList[] = {};
        QuickSort quickSort = new QuickSort();
        quickSort.sort(list, 0, 0);
        assert (Arrays.equals(list, expectedList));
    }
}
